### DISCUSSION TITLE
Speech recognition polyfill and SAL configuration

### DISCUSSION THREAD

**Discussion by jsjf93** [speechly/speechly#214] [Q&A] [answered]
Hi all,

Got a couple of questions, and apologies in advance if I'm being daft!

1) Do entities defined in the SAL configuration work with the speech recognition polyfill?
2) If yes to the above, how would I go about training the model on a domain specific term such as "Uppr" (like upper)

In my configuration, I put in a few examples like these:
```
*select Show me [Uppr] details
*select Get [Uppr] company address
*select Is [Uppr] a good business
```

But I'm only getting the regular spelling of the word "upper", and now wondering if I've misunderstood something or perhaps there needs to be more training data?

Any help would be greatly appreciated.

Many thanks
Josh

p.s. I started using Speechly today, it is a fantastic product, many thanks to everyone involved in making it.

**Comment by bigdatabaracus** [comment, marked as answer]
Hello @jsjf93, thanks for the good question! 

1. unfortunately the entities and intents are not supported by the speech recognition polyfill. If you need that functionality you need to take into use either the Speechly [browser](https://github.com/speechly/speechly/tree/main/libraries/browser-client) or [React](https://github.com/speechly/speechly/tree/main/libraries/react-client) client. Fortunately they should be pretty easy to use. 

2. In principal your configuration should work as you expect. However, the word "upper" is such common word that you would need quite a bit more data to bias the speech recognition enough. If you have a fairly simple application, with relatively simple set of commands, that might be hard to do. What you can do however, is to use the entity tagging in combination our [lookup entity data type](https://docs.speechly.com/slu-examples/imports-and-lookups/#the-lookup-entity-data-type).  

In this case you configuration would look something like this: 

<img width="836" alt="image" src="https://user-images.githubusercontent.com/4426400/181235632-467b2704-95e7-4e9e-8c26-d6dd6111ddaa.png">

Pressing the  <img width="34" alt="image" src="https://user-images.githubusercontent.com/4426400/181235349-dd5047ec-27d0-41bc-9582-f06b91ba9cdf.png"> icon in your entity list you can define you lookup table to look something like this:

<img width="836" alt="image" src="https://user-images.githubusercontent.com/4426400/181235714-f0e19cb6-a866-49c0-85ac-c385f3922a37.png">

You will find that the entity value is now formatted as you wanted it: 

![image](https://user-images.githubusercontent.com/4426400/181236464-8b893ef5-5057-4258-ad60-f073fb9644d5.png)

Hope this helped Josh and thank you for the kind words about the product. 

best regards, -Hannes-

**Reply by jsjf93** [reply]
Thanks very much @bigdatabaracus - that is really helpful
